waiter/src/waiter/token.clj

```diff
@@ -516,48 +516,64 @@
                            :log-level :warn}))))
 
       nil
-      (do
-        (when (and run-as-user (not= "*" run-as-user))
+      (let [existing-editor (get existing-token-metadata "editor")
+            existing-owner (get existing-token-metadata "owner")
+            creating-token? (empty? existing-token-metadata)
+            existing-owner? (and existing-owner
```

Comment:
Nit-pick: One might confuse the meaning of `existing-owner?` with the meaning "token has an existing owner".  Here it means "token has an existing owner and the authenticated-user can be that owner"

Reply:
Changed to `current-owner?`